### PR TITLE
Add ValueFromPipelineByPropertyName to Set-PSResourceRepository

### LINKED ISSUE #670: Set-PSResourceRepository does not accept pipeline input from Get-PSResourceRepository
### Prerequisites

- [X] Write a descriptive title.
- [X] Make sure you are able to repro it on the [latest released version](https://www.powershellgallery.com/packages/PowerShellGet)
- [X] Search the existing issues.

### Steps to reproduce

I cannot pipe the output from `Get-PSResouceRepository` to `Set-PSResourceRepository`. The error message that is thrown is also not something that I would expect. This issue can be fixed by adding `ValueFromPipelineByPropertyName` on the `Name` parameter.

### Expected behavior

```console
PS /> Get-PSResourceRepository | Set-PSResourceRepository -Trusted
```

### Actual behavior

```console
PS /> Get-PSResourceRepository | Set-PSResourceRepository -Trusted
Set-PSResourceRepository: Updating to repository store failed: Cannot find the repository because it does not exist. Try registering the repository using 'Register-PSResourceRepository'
PS /> Get-PSResourceRepository

Name      Uri                                      Trusted Priority
----      ---                                      ------- --------
PSGallery https://www.powershellgallery.com/api/v2 False   50

PS /> Set-PSResourceRepository -Name PSGallery -Trusted
```

### Error details

```console
PS /> Get-Error

Exception             :
    Type        : System.Management.Automation.PSInvalidOperationException
    ErrorRecord :
        Exception             :
            Type    : System.Management.Automation.ParentContainsErrorRecordException
            Message : Updating to repository store failed: Cannot find the repository because it does not exist. Try registering the reposi
tory using 'Register-PSResourceRepository'
            HResult : -2146233087
        CategoryInfo          : InvalidOperation: (:) [], ParentContainsErrorRecordException
        FullyQualifiedErrorId : InvalidOperation
    TargetSite  :
        Name          : ThrowTerminatingError
        DeclaringType : System.Management.Automation.MshCommandRuntime, System.Management.Automation, Version=7.2.4.500, Culture=neutral, PublicKe
yToken=31bf3856ad364e35
        MemberType    : Method
        Module        : System.Management.Automation.dll
    Message     : Updating to repository store failed: Cannot find the repository because it does not exist. Try registering the repository
 using 'Register-PSResourceRepository'
    Source      : System.Management.Automation
    HResult     : -2146233079
    StackTrace  :
   at System.Management.Automation.MshCommandRuntime.ThrowTerminatingError(ErrorRecord errorRecord)
TargetObject          : Microsoft.PowerShell.PowerShellGet.Cmdlets.SetPSResourceRepository
CategoryInfo          : InvalidArgument: (Microsoft.PowerShel…SResourceRepository:SetPSResourceRepository) [Set-PSResourceRepository], PSInvalidOp
erationException
FullyQualifiedErrorId : ErrorInNameParameterSet,Microsoft.PowerShell.PowerShellGet.Cmdlets.SetPSResourceRepository
InvocationInfo        :
    MyCommand        : Set-PSResourceRepository
    ScriptLineNumber : 1
    OffsetInLine     : 28
    HistoryId        : 14
    Line             : Get-PSResourceRepository | Set-PSResourceRepository -Trusted
    PositionMessage  : At line:1 char:28
                       + Get-PSResourceRepository | Set-PSResourceRepository -Trusted
                       +                            ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    InvocationName   : Set-PSResourceRepository
    CommandOrigin    : Internal
ScriptStackTrace      : at <ScriptBlock>, <No file>: line 1
```

### Environment data

```powershell
PS /> Get-Module PowerShellGet

ModuleType Version    PreRelease Name                                ExportedCommands
---------- -------    ---------- ----                                ----------------
Binary     3.0.14     beta14     PowerShellGet

PS /> Get-Module PowerShellGet

ModuleType Version    PreRelease Name                                ExportedCommands
---------- -------    ---------- ----                                ----------------
Binary     3.0.14     beta14     PowerShellGet
```

### Visuals

_No response_

### PR DESCRIPTION
**Pull request by alerickson**
# PR Summary
Adding the property `ValueFromPipelineByPropertyName` to the -Path parameter in Set-PSResourceRepository so that pipeline input can be used to set a respository, eg:

`Get-PSResourceRepository "TestRepo" | Set-PSResourceRepository -Trusted`

## PR Context

Resolves #670 

## PR Checklist

- [x] [PR has a meaningful title](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
    - Use the present tense and imperative mood when describing your changes
- [x] [Summarized changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] [Make sure all `.h`, `.cpp`, `.cs`, `.ps1` and `.psm1` files have the correct copyright header](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] This PR is ready to merge and is not [Work in Progress](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---work-in-progress).
    - If the PR is work in progress, please add the prefix `WIP:` or `[ WIP ]` to the beginning of the title (the `WIP` bot will keep its status check at `Pending` while the prefix is present) and remove the prefix when the PR is ready.
- **[Breaking changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#making-breaking-changes)**
    - [x] None
    - **OR**
    - [x] [Documentation needed]()
        - [ ] Issue filed: 
- **User-facing changes**
    - [x] Not Applicable
    - **OR**
    - [ ] [Documentation needed]()
        - [ ] Issue filed: 
- **Testing - New and feature**
    - [x] N/A or can only be tested interactively
    - **OR**
    - [ ] [Make sure you've added a new test if existing tests do not effectively test the code changed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#before-submitting)
- **Tooling**
    - [x] I have considered the user experience from a tooling perspective and don't believe tooling will be impacted.
    - **OR**
    - [ ] I have considered the user experience from a tooling perspective and enumerated concerns in the summary.

### CODE REVIEW

**Comment by alerickson** [comment]
Ready for review.

File: src/code/SetPSResourceRepository.cs [hunk 1]
```diff
@@ -34,7 +34,7 @@ public sealed class SetPSResourceRepository : PSCmdlet
         /// <summary>
         /// Specifies the name of the repository to be set.
         /// </sumamry>
-        [Parameter(Mandatory = true, Position = 0, ValueFromPipeline = true, ParameterSetName = NameParameterSet)]
+        [Parameter(Mandatory = true, Position = 0, ValueFromPipeline = true, ValueFromPipelineByPropertyName = true, ParameterSetName = NameParameterSet)]
```
Comment (anamnavi) [inline src/code/SetPSResourceRepository.cs:37]: Can you also make note of this in the help doc, for the Name property where it says accepts pipeline input we currently only have `True (byValue)` but would also need to signify by property name. Thanks!

**Review by anamnavi — APPROVED** [review]
LGTM, but if we can also update the help doc that would be great, thanks!

File: src/code/SetPSResourceRepository.cs (hunk 1 above)
Comment (alerickson) [reply src/code/SetPSResourceRepository.cs:37]: I think if I add this now it'll get overwritten with Sean's changes, so I'll just leave a comment on his PR either asking him to do it or as a reminder that I should add that in.  Thank you for flagging this though!

**Pull request merged**